sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -853,6 +853,19 @@ object IntervalUtils {
     new CalendarInterval(totalMonths, totalDays, micros)
   }
 
+  def makeMicrosInterval(
+      days: Int,
+      hours: Int,
+      mins: Int,
+      secs: Decimal): Long = {
+    assert(secs.scale == 6, "Seconds fractional must have 6 digits for microseconds")
+    var micros = secs.toUnscaledLong
+    micros = Math.addExact(micros, Math.multiplyExact(days, MICROS_PER_DAY))
+    micros = Math.addExact(micros, Math.multiplyExact(hours, MICROS_PER_HOUR))
```

Comment:
Should hours (and other fields) be in some range? See the table from the SQL:2016
<img width="595" alt="Screenshot 2021-06-14 at 20 41 45" src="https://user-images.githubusercontent.com/1580697/121935371-07b0bb80-cd51-11eb-851d-000c865f99c4.png">
If not, please, explain.